app/controllers/answers_controller.rb

```diff
@@ -0,0 +1,56 @@
+class AnswersController < ApplicationController
+  before_action :set_answer, only: [:show, :edit, :update, :destroy]
+  before_action :authenticate_user!
+  authorize_resource
+  # GET /answers/1/edit
+  def edit
+  end
+  
+  # POST /answers
+  # POST /answers.json
+  def create
+    #binding.pry
+    @answer  = current_user.answers.build(answer_params)
+    
+    respond_to do |format|
+      if @answer.save
+        format.html { redirect_to request.referer}
+      else
+        format.html { render action: 'new' }
+      end
+    end
+  end
+
+  # PATCH/PUT /answers/1
+  # PATCH/PUT /answers/1.json
+  def update
+    respond_to do |format|
+      if @answer.update(answer_params)
+        format.html { redirect_to Question.find(@answer.question_id), notice: 'Answer was successfully updated.' }
```

Comment:
You should not load an association using `find` and association's ID. i.e. here you should do the following.
`@answer.question`